### ISSUE TITLE
Setting clone_depth: 1, not working

### ISSUE THREAD

**Issue by Rhymond** [Block8/PHPCI#1141]
In phpci.yml i set:

    build_settings:
        clone_depth: 1

But when i watch server processes i see:
git clone --recursive -b master git@bitbucket.org:user/my-project.git /var/www/phpci/PHPCI/build/754_41edd/

Where is --depth 1

**Comment by REBELinBLUE** [comment]
Hmm, odd https://github.com/Block8/PHPCI/blob/92ee90998a1b6b218656195a78366542d74a97a3/PHPCI/Model/Build/RemoteGitBuild.php#L92 it should be between`--recursive` and `-b`

I'll have a play around later and see if I can figure out what is going on. Are you running from `master` or from a tag?

**Comment by Rhymond** [comment]
Master.
In Model/build/RemoteGitBuild.php file function cloneBySsh().

    $depth = $builder->getConfig('clone_depth');

    if (!is_null($depth)) {
        $cmd .= ' --depth ' . intval($depth) . ' ';
    }

can't get config **clone_depth**
When i set it manualy in code:

    $depth = 2;

    if (!is_null($depth)) {
        $cmd .= ' --depth ' . intval($depth) . ' ';
    }

It works perfectly.

**Comment by REBELinBLUE** [comment]
@dancryer Have there been any changes to the builder which could cause this that you are aware of? I know you were doing branch specific settings so maybe that?

**Comment by Adirelle** [comment]
Look at [RemoteGitBuild::createWorkingCopy](https://github.com/Block8/PHPCI/blob/master/PHPCI/Model/Build/RemoteGitBuild.php#L34): the repositoriy is cloned before the configuration is loaded, so there is no way cloneDepth could work with RemoteGitBuild.

**Comment by REBELinBLUE** [comment]
Hmm yeah, looking at the history it doesn't look like it could have ever worked.....

In fact, thinking about it, I don't see how it could have even been expected to work, because you'd have to clone the repository first to get the `phpci.yml` file in order to read the `clone_depth` setting.

**Comment by Adirelle** [comment]
IMO this should work the other way around: always do shallow clones (with --depth 1), read the configuration file, then pull missing commits using "git pull --unshallow" or "git pull --depth".

**Comment by REBELinBLUE** [comment]
Yeah that would be the logical way, or have it in the project config in the DB

**Comment by Rhymond** [comment]
@Adirelle i think this is better solution to download only configuration file
git archive --remote=git://remote/project.git HEAD phploy.yml | tar -x
